test/fakeOrigin/endpoint/endpoint.go

```diff
@@ -0,0 +1,209 @@
+	for k, v := range tokenDict {
+		input = strings.Replace(input, k, v, -1)
+	}
+	return input
+}
+
+// DefaultConfig generates a basic default configuration
+func DefaultConfig() Config {
+	return Config{
+		ServerConf: ServerInfo{
+			HTTPListeningPort:  8080,
+			HTTPSListeningPort: 8443,
+			SSLCert:            DefaultHTTPSCertFile,
+			SSLKey:             DefaultHTTPSKeyFile,
+			BindingAddress:     "",
+			CrossdomainFile:    "./example/crossdomain.xml",
+		},
+		Endpoints: []Endpoint{
+			{
+				ID:              "SampleFile",
+				Source:          "./example/video/kelloggs.mp4",
+				OutputDirectory: DefaultOutputDirectory,
+				EndpointType:    Static,
+			},
+			{
+				ID:              "SampleDirectory",
+				Source:          "./example/video/",
+				OutputDirectory: DefaultOutputDirectory,
+				EndpointType:    Dir,
+			},
+		},
+	}
+}
+
+// WriteConfig saves a fakeOrigin config as a pretty-printed json file
+func WriteConfig(cfg Config, path string) error {
+	bts, err := json.MarshalIndent(cfg, "", "\t")
+	if err != nil {
+		return errors.New("marshalling JSON: " + err.Error())
+	}
+	if err = ioutil.WriteFile(path, bts, 0755); err != nil {
+		return errors.New("writing file '" + path + "': " + err.Error())
+	}
+	return nil
+}
+
+// ProcessConfig processes the config loaded from disk, or generated the first time. This must be called before the config can be used to transcode or serve.
+func ProcessConfig(out Config) Config {
+	for i := range out.Endpoints {
+		// Resolve relative paths to absolute paths
+		out.Endpoints[i].Source, _ = filepath.Abs(out.Endpoints[i].Source)
+		out.Endpoints[i].OutputDirectory, _ = filepath.Abs(out.Endpoints[i].OutputDirectory)
+
+		if out.Endpoints[i].OutputDirectory == "" {
+			out.Endpoints[i].OutputDirectory = DefaultOutputDirectory
+		}
+		if out.Endpoints[i].DiskID == "" {
+			out.Endpoints[i].DiskID = out.Endpoints[i].ID
+		}
+	}
+	return out
+}
+
+// LoadAndGenerateDefaultConfig loads the config from a given json file and puts a default value in place if you havn't stored anything
+func LoadAndGenerateDefaultConfig(path string) (Config, error) {
+	out := DefaultConfig()
+	defaultEndpoints := make([]Endpoint, len(out.Endpoints))
+	copy(defaultEndpoints, out.Endpoints)
+	raw, err := ioutil.ReadFile(path)
+	if err != nil || len(raw) == 0 {
+		raw = []byte("{}")
+	}
+	err = json.Unmarshal(raw, &out)
+	if err != nil {
+		return out, err
+	}
+	if err := WriteConfig(out, path); err != nil {
+		return out, errors.New("writing config to file: " + err.Error())
+	}
+	if fmt.Sprintf("%v", out.Endpoints) == fmt.Sprintf("%v", defaultEndpoints) {
+		return out, errors.New("default endpoints generated, please provide real input")
+	}
+	for _, ep := range out.Endpoints {
+		if len(ep.ManualCommand) > 0 {
+			//if !contains(ep.ManualCommand, "%MASTERMANIFEST%") {
+			//	return out, errors.New("Manual commands must include the %MASTERMANIFEST% token")
+			//}
+			if !contains(ep.ManualCommand, "%OUTPUTDIRECTORY%") {
```

Comment:
this isn't a huge deal, but some syntax highlighting will mark anything after a `%` as a replacement character - even when there's no context for that. If you use grave accents (<kbd>`</kbd>) you can disable that erroneous behavior.

Reply:
Addressed with new commits